src/framework/dockwindow/thirdparty/KDDockWidgets/src/private/FloatingWindow.cpp

```diff
@@ -399,7 +399,7 @@ void FloatingWindow::updateTitleBarVisibility()
     if (KDDockWidgets::usesClientTitleBar()) {
         const auto flags = Config::self().flags();
         if ((flags & Config::Flag_HideTitleBarWhenTabsVisible) && !(flags & Config::Flag_AlwaysTitleBarWhenFloating)) {
-            if (hasSingleFrame()) {
+            if (!frames().isEmpty() && hasSingleFrame()) {
```

Comment:
It's best not to call frames() more than once. Looking at its implementation, it's not free